File: Position_Salaries.R
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

lin_reg = lm(formula = Salary ~ .,
             data = dataset)

dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
poly_reg = lm(formula = Salary ~.,
              data = dataset)

#Linear Regression Model
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'Red') +
  geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = dataset)),
            color = 'Blue') +
  ggtitle('Salary vs Level') +
  xlab('Levels') +
  ylab('Salary')

#Polynomial Regression Model
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'Red') +
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)),
            color = 'Blue') +
  ggtitle('Salary vs Levels') +
  xlab('Levels') +
  ylab('Salary')

#Predicting Salary at Level = 6.5
y_pred = predict(lin_reg, data.frame(Level = 6.5))

y_poly_pred = predict(poly_reg, data.frame(Level = 6.5, Level2 = 6.5^2, Level3 = 6.5^3))

File: Position_Salaries.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, 2].values

from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X, y)

from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)
lin_reg_2 = LinearRegression()
lin_reg_2.fit(X_poly, y)

plt.scatter(X, y, color = 'red')
plt.plot(X, lin_reg.predict(X), color = 'blue')
plt.title('Level vs Salary(Linear Regression)')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.show()

plt.scatter(X, y, color = 'red')
plt.plot(X, lin_reg_2.predict(X_poly), color = 'blue')
plt.title('Level vs Salary(Polynomial Regression)')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.show()

#For a more continuous graph of polynomial regression
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, lin_reg_2.predict(poly_reg.fit_transform(X_grid)), color = 'blue')
plt.title('Level vs Salary(Polynomial Regression)')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.show()

#Prediction with Linear Regression
lin_reg.predict(6.5)

#Prediction with Polynomial Regression
lin_reg_2.predict(poly_reg.fit_transform(6.5))